### ISSUE TITLE
Setting pointcut to SCOPE.METHODS does not work.

### ISSUE THREAD

**Issue by Rob-pw** [amoilanen/jsAspect#17]
In all test cases, using the examples provided, the pointcut SCOPE.METHODS does not work. 

``` javascript
var jsAspect = require('js-aspect');

var afterAdviceBehaviour = function() {
    console.log("joinPoint", "after");
};
var afterAdvice =  new jsAspect.Advice.After(afterAdviceBehaviour, jsAspect.SCOPE.METHODS); //set the pointcut

//Define behaviour of the beforeAdvice
var beforeAdviceBehaviour = function() {
    console.log("joinPoint", "before");
};
var beforeAdvice = new jsAspect.Advice.Before(beforeAdviceBehaviour, jsAspect.SCOPE.METHODS); //set the pointcut

//create the aspect
var aspect = new jsAspect.Aspect(beforeAdvice, afterAdvice);

function Target() {
    this.method1 = function() {
        return "method1value";
    };
}

var target = new Target();

aspect.applyTo(target);

target.method1(); // Nothing, no console logs.

var foo = {
    method1: function () {
        return "foobar.";
    }
};

aspect.applyTo(foo);

foo.method1(); // Nothing, no console logs.
```

Very simple example, but not working as expected.

Am I doing something wrong?

**Comment by Rob-pw** [comment]
Curiously the following does work:

``` javascript
jsAspect.inject(foo, jsAspect.SCOPE.METHODS, jsAspect.JOIN_POINT.AFTER_RETURNING, function (context) {
    console.log("Before", context);
});
```

**Comment by amoilanen** [comment]
I will take a look, can be a regression after some re-factoring.

**Comment by amoilanen** [comment]
The problem is actually in the outdated documentation. In the current version it was not possible to pass scope as an argument to the advice constructor. Instead the method `withPointcut` should be used. Rather than:

``` javascript
    var afterAdviceBehaviour = function() {
      console.log("joinPoint", "after");
    };
    var afterAdvice =  new jsAspect.Advice.After(afterAdviceBehaviour, jsAspect.SCOPE.METHODS);
```

it should be

``` javascript
    var afterAdviceBehaviour = function() {
      console.log("joinPoint", "after");
    };
    var afterAdvice =  new jsAspect.Advice.After(afterAdviceBehaviour);

    afterAdvice.withPointcut(jsAspect.SCOPE.METHODS);
```

However, since at least two people (author of the documentation and you) see passing scope definition as an argument to constructor as a natural API to use I think it makes sense to support it. 

Implemented in af5dfef780bee4cf649edb144302c9973cd00d80, I also added the code provided in this issue as two additional test cases.

It also makes sense to improve the documentation which I will probably do as well. The source of confusion here was the documentation, sorry for the inconvenience and thank you for reporting the issue.
